WordPress/src/main/java/org/wordpress/android/viewmodel/themes/ThemesViewModel.kt

```diff
@@ -0,0 +1,107 @@
+import javax.inject.Named
+
+class ThemesViewModel @Inject constructor(
+    private val themeStore: ThemeStore,
+    private val resourceProvider: ResourceProvider,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
+    private lateinit var site: SiteModel
+    private var isStarted = false
+
+    private val _bottomSheetUiState = MutableLiveData<BottomSheetUIState>()
+    val bottomSheetUiState: LiveData<BottomSheetUIState> = _bottomSheetUiState
+
+    private val _bottomSheetAction = MutableLiveData<BottomSheetAction>()
+    val bottomSheetAction: MutableLiveData<BottomSheetAction> = _bottomSheetAction
+
+    fun start(site: SiteModel) {
+        if (isStarted) return
+
+        this.site = site
+        _bottomSheetUiState.value = BottomSheetUIState()
+        isStarted = true
+    }
+
+    private fun currentViewState(): BottomSheetUIState = bottomSheetUiState.value!!
+
+    fun onActivateMenuItemClicked(themeId: String) {
+        if (!site.isUsingWpComRestApi) {
+            AppLog.i(THEMES, "Theme activation requires a site using WP.com REST API. Aborting request.")
+            return
+        }
+
+        val theme = themeStore.getInstalledThemeByThemeId(site, themeId) ?: themeStore.getWpComThemeByThemeId(themeId)
+        if (theme == null) {
+            AppLog.w(THEMES, "Theme unavailable to activate. Fetch it and try again.")
+            return
+        }
+
+        bottomSheetAction.value = Show
+        setBottomSheetTexts(theme.name)
+    }
+
+    fun onUseThemeHomepageSelected() = setBottomSheetSelection(UseThemeHomepage)
+
+    fun onKeepCurrentHomepageSelected() = setBottomSheetSelection(KeepCurrentHomepage)
+
+    fun onPreviewButtonClicked() {
+        // todo
+    }
+
+    fun onActivateButtonClicked() {
+        // todo
+    }
+
+    fun onDismissButtonClicked() {
+        bottomSheetAction.value = Hide
+    }
+
+    private fun setBottomSheetTexts(themeName: String) {
+        _bottomSheetUiState.value = currentViewState().copy(
+                themeNameText = themeName,
+                sheetInfoText = resourceProvider.getString(R.string.theme_bottom_sheet_info, themeName),
+                useThemeHomePageOptionText = resourceProvider
+                        .getString(R.string.theme_bottom_sheet_use_theme_layout_main_text, themeName)
+        )
+    }
+
+    private fun setBottomSheetSelection(selection: Selection) {
+        _bottomSheetUiState.value = currentViewState().copy(selection = selection)
+    }
+
+    sealed class BottomSheetAction {
+        object Show : BottomSheetAction()
+        object Hide : BottomSheetAction()
+    }
+
+    data class BottomSheetUIState(
+        val themeNameText: String = "",
+        val sheetInfoText: String = "",
+        val useThemeHomePageOptionText: String = "",
+        val selection: Selection = KeepCurrentHomepage
+    ) {
+        sealed class Selection {
+            object UseThemeHomepage : Selection()
```

Comment:
This approach works but looks like it’s going to require to [add the show/hide logic in the view](https://github.com/wordpress-mobile/WordPress-Android/pull/15043/files#diff-b63984509c186a64946e6e300e9b241a0f70f13a13eb0de5c673bdf8435c6fd3R54).  I think it might eliminate this and keep this under the control of the view model by adding the visibility state in the sealed class. Eg. the `Selection ` may have two constants like `sealed class Selection(open val homePageCheckmarkVisible: Boolean = false, open val keepCurrentCheckmarkVisible: Boolean = false)`. We can then set the desired visibility for each state (e.g. `object UseThemeHomepage : Selection(homePageCheckmarkVisible = true, keepCurrentCheckmarkVisible = false)`). Our view/fragment can then be changed to just apply the UiState without having any knowledge of the logic behind it.
([example](https://github.com/wordpress-mobile/WordPress-Android/blob/70ce7de8817020d29768074bb408688436320220/WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerUiState.kt))

Reply:
In that case, wouldn't it be better to use `View.VISIBLE` and `View.INVISIBLE` instead of `true` and `false`?

Reply:
That can work too. My personal preference would be the `boolean` value to avoid using View specific things in the Model (thought it is actually an `Int`).

Reply:
This is addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/15043/commits/64637e604557b1bb5ff0cb7aa99cc4779b5e6455, and tests are modified in https://github.com/wordpress-mobile/WordPress-Android/pull/15043/commits/32e90eccb38fb2412bad21e6ea05b99026dee005. I did not keep the `Selection` sealed class and move the option to parent class along with other ui state options like texs.

Reply:
This looks good @kashandanish 👍 
Thank you for making the suggested changes.